### ISSUE TITLE
Renaming a file without permission - the file cannot be synced anymore 

### ISSUE THREAD

**Issue by HanaGemela** [owncloud/client#7490]
2.6.0rc1 (build 12411), mac 10.14.6:

Steps to recreate:

1. User1 shares a folder with a group without change and create permissions. The folder includes one file
2. User2 (member of that group) renames the file

Actual result: The error message is 'no permission to add files'. The error message should state that rename is not allowed and the file should be either renamed back to the original name so it is still synced or the message should mention the original name and if the user renames the file back, it gets synced again. Currently renaming to the original file name causes the same 'not allowed to add new files' error.
Basically the user doesn't know that they are not allowed to rename files. And once they do it, the file is not syncing anymore and there is no way back.

(From comment https://github.com/owncloud/client/issues/7410#issuecomment-529464006 , contains a screenshot)

**Comment by ogoffart** [comment]
It would be easy to change the error message to 

> Not allowed because you don't have permission to add **or move** files in that folder

As for the restoration, we do not currently rename the file back its original name, but we still restore (download) the file with the old name without deleting the file with the new name, for fear of data loss.

I am not convinced it is a good idea to simply restore the file by renaming it.

We actually had an option to do it back in the days and up to 2.5: It was done with commit 5676685f5845b317745d6ff8e7561537c9414a13 which explain in the comments why we do not do it by default.

Any opinions on what should we do here?

**Comment by HanaGemela** [comment]
Redownloading the file is OK. Comments suggest that that should be the default behaviour. But it is not

**Comment by ogoffart** [comment]
> Redownloading the file is OK. Comments suggest that that should be the default behaviour. But it is not

It does that for me.
The file with the old name is restored (redownloaded) and the file with the new name is still there with an error.
Is that not the expected behavior?

**Comment by HanaGemela** [comment]
Redownload works again in 2.6.0-daily20191009 (build 12541), macOS 10.15.

The error message is still wrong.

Not sure about 

> and the file with the new name is still there with an error.

@michaelstingl @pmaier1 what is the expected behaviour?

**Comment by TheOneRing** [comment]
Duplicate of: https://github.com/owncloud/client/issues/7413
